changelog/620.bugfix

```diff
@@ -0,0 +1,3 @@
+Use the execnet main_thread_only execmodel so that code which expects to run in the main thread will just work.
+
+Also increase minimum python version to 3.8 since execnet dropped 3.7 support in pytest-dev/execnet#245.
```

Comment:
```suggestion
Use the ``execnet`` new ``main_thread_only`` "execmodel" so that code which expects to only run in the main thread will now work as expected.
```

Let's mention the Python 3.7 drop into its own note. :+1: